### ISSUE TITLE
Is there a built-in way to load a saved model to use for other networks?

### ISSUE THREAD

**Issue by nramani611** [bowang-lab/BIONIC#36]
I am running bionic and able to save the model to a pt file. Is there a built-in way to load the BIONIC model with the saved pt file to use on other networks. If not, what is the easiest way to accomplish this? I am using the wheel instillation of bionic.

**Comment by duncster94** [comment]
Hi, check this out:
https://github.com/bowang-lab/BIONIC/blob/fa9f292f32004eaaaa285bfc63df517124102b35/bionic/train.py#L138-L141

You can try adding `{"load_pretrained_model": true}` to the config and make sure `out_name` is the same name as your model (minus the "_model.pt") at the end. It is expected that the saved model is in the "models" subdirectory which could be a problem if you're using the wheel version.

The above implementation is left over from a much earlier version of BIONIC so there's a good chance it wont work. Give it a try and let me know if you have any success. I haven't got around to supporting pretrained models yet but if you want I can implement, test and push a new BIONIC version later this week.

**Comment by nramani611** [comment]
I also have the Poetry version installed, so I can try it there, but I am confused about the "models" directory. Is this a directory somewhere within the cloned files, or should I be making a "models" directory at a specific location and storing the pt file there?

**Comment by duncster94** [comment]
If you have the Poetry version you can just change `f"models/{self.params.out_name}_model.pt"` to point to the directory where you've saved your models.

**Comment by nramani611** [comment]
I was able to make modify the code so that it uses the pretrained model, but it still trains the model, meaning the weights are changing. Is there a way to disable this and just get the embeddings from the autoencoder?

**Comment by duncster94** [comment]
As a quick hack you could just set the epochs to 0. If that doesn't work than you could set the learning rate to 0 and the epochs to 1.

**Comment by nramani611** [comment]
This seems to work. Thanks for your help!

**Comment by nramani611** [comment]
Hello, I am reopening this issue as I am trying to load a pretrained model that was trained with labels. However, when I try to load this saved model with different inputs, I get the following error:
![image](https://user-images.githubusercontent.com/43153496/191090719-71538d9c-03f8-42f0-b43e-fac83ccdca52.png)

**Comment by duncster94** [comment]
Hi, it looks like the classification head is missing from the saved model. Try the following and let me know the output:

    state_dict = torch.load("path/to/saved/model.pt")
    print([key for key in state_dict.keys() if "Classification" in key])

Also, did you run the original model with `model_parallel = True` and the current model with `model_parallel = False` or vice versa?

**Comment by nramani611** [comment]
Here is the output. The original model was ran with model_parallel = True. I just tried rerunning the current model with model_parallel = False, and I got the same error posted above.
![image](https://user-images.githubusercontent.com/43153496/191096073-18a9ce7c-62ee-47ad-b5f6-c7067c063396.png)

**Comment by duncster94** [comment]
I've tried reproducing the error but was unable to. Does the current model that you're loading the pretrained weights into have a classification head (i.e. are you providing a label file in "label_names" for the new model)? 

If this is the case and you want to use the encoder weights but not the classification head, you can modify the saved state dictionary in a separate script by deleting the "Classification" tensors. You should then be able to load the new state dictionary into your script like normal.

EDIT: Or just add the kwarg `strict=False` to the `load_state_dict` function call.

**Comment by nramani611** [comment]
I just realized that the model was not saved to the proper place, meaning I was calling the old pt file that was generated from unsupervised learning. I moved the file and it works now.
